File: ZoneSentry/Models/RentAgreement.cs
namespace ZoneSentry.Models;

public class RentAgreement
{
    public int Id { get; set; }
    
    public DateTime Date { get; set; }
    public DateTime ExpirationDate { get; set; }
    public int PaymentPerMonth { get; set; }
    
    public Realty Realty { get; set; }
    public ApplicationUser? Owner { get; set; }
    public bool OwnedByCompany { get; set; }
    public ApplicationUser Tenant { get; set; }
    
    public List<RentPayment> Payments { get; set; }
}

public class RentAgreementUserView
{
    public int Id { get; set; }
    
    public DateTime Date { get; set; }
    public DateTime ExpirationDate { get; set; }
    public int PaymentPerMonth { get; set; }
    
    public RealtyUserView Realty { get; set; }
    public ApplicationUserDTO? Owner { get; set; }
    public bool OwnedByCompany { get; set; }
    public ApplicationUserDTO Tenant { get; set; }
}

public class RentAgreementDTO
{
    public int Id { get; set; }
    
    public DateTime Date { get; set; }
    public DateTime ExpirationDate { get; set; }
    public int PaymentPerMonth { get; set; }
    
    public bool OwnedByCompany { get; set; }
    public ApplicationUserDTO Tenant { get; set; }
}

File: ZoneSentry/ClientApp/src/api/models/RentAgreement.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ApplicationUser } from './ApplicationUser';
import type { Realty } from './Realty';
import type { RentPayment } from './RentPayment';

export type RentAgreement = {
    id?: number;
    date?: string;
    expirationDate?: string;
    paymentPerMonth?: number;
    realty?: Realty;
    owner?: ApplicationUser;
    ownedByCompany?: boolean;
    tenant?: ApplicationUser;
    payments?: Array<RentPayment> | null;
};